app/models/manageiq/providers/vmware/infra_manager/refresh_parser/filter.rb

```diff
@@ -21,6 +21,8 @@ def filter_vc_data(ems, target)
           filtered_data[:folder], filtered_data[:dc], filtered_data[:cluster], filtered_data[:host_res] =
             ems_metadata_inv_by_host_inv(host_data, vm_data)
           filtered_data[:rp] = rp_inv_by_host_inv(host_data)
+          filtered_data[:storage_profile], filtered_data[:storage_profile_datastore], filtered_data[:storage_profile_entity] =
+            storage_profile_inv_by_host_inv(host_data)
```

Comment:
Returning all profiles fixes the bug but I think there is more filtering we can do here